File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeNthFromEnd(head *ListNode, n int) *ListNode {
    fast := head
    slow := head
    
    for i := 0; i < n; i++ {
        fast = fast.Next
    }
    
    if fast == nil {
        return head.Next
    }
    
    for fast.Next != nil {
        fast = fast.Next
        slow = slow.Next
    }
    
    slow.Next = slow.Next.Next
    
    return head
}

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution 
{
    public ListNode RemoveNthFromEnd(ListNode head, int n) 
    {
        ListNode fast = head;
        ListNode slow = head;
        
        for (int i = 0; i < n; i++) 
        {
            fast = fast.next;
        }
        
        if (fast == null)
        {
            return head.next;
        }
        
        while (fast.next != null) 
        {
            fast = fast.next;
            slow = slow.next;
        }
        
        slow.next = slow.next.next;
        
        return head;
    }
}